test/index.js

```diff
@@ -9,7 +9,9 @@ test('Tests run', (assert) => {
   assert.end();
 });
 
-test('Greet World', (assert) => {
+
+test('Greet World', (assert) => new Promise((resolve) => {
```

Comment:
That is expected behaviour, not an issue. Returning a `new Promise` does not invoke constructor for side-effects only (what actually is checked by `no-new` rule http://eslint.org/docs/rules/no-new)